test/common/upstream/subset_lb_test.cc

```diff
@@ -936,6 +956,53 @@ TEST_P(SubsetLoadBalancerTest, OnlyMetadataChanged) {
   EXPECT_EQ(host_set_.hosts_[1], lb_->chooseHost(&context_13));
 }
 
+TEST_P(SubsetLoadBalancerTest, EmptySubsetsGetPurged) {
+  std::vector<SubsetSelectorPtr> subset_selectors = {makeSelector({"version"}),
+                                                     makeSelector({"stage"})};
```

Comment:
Did you mean to make the second selector `makeSelector({"version", "stage"})`?

That way the version subset has children. As defined, the subsets are always either based on just version or just stage (though host[1] appears in multiple subsets).

In addition, I think two more cases might be useful. I suppose I would look at the coverage report and make sure all the new code has coverage. 

1. update host[1] to be version="1.2" (e.g. same as host[0]) and stage="prod", then remove stage="prod", and then add it back.

2. update host[1] to be version="1.2" (e.g. same as host[0]) and stage="prod", and then change both host[0] and host[1] to have a different version.

Reply:
I actually did mean `makeSelector({"version", "stage"})` -- thanks for catching that. Fixing.

Reply:
Fixed